contentctl/objects/content_versioning_service.py

```diff
@@ -0,0 +1,548 @@
+import time
+import uuid
+import json
+import re
+import logging
+from typing import Any, Callable
+from functools import cached_property
+
+from pydantic import BaseModel, PrivateAttr, computed_field, Field
+import splunklib.client as splunklib                                                                # type: ignore
+from splunklib.binding import HTTPError, ResponseReader                                             # type: ignore
+from splunklib.data import Record                                                                   # type: ignore
+
+from contentctl.objects.config import test_common, Infrastructure
+from contentctl.objects.detection import Detection
+from contentctl.objects.correlation_search import ResultIterator
+from contentctl.helper.utils import Utils
+
+# TODO (cmcginley):
+#   - [x] version naming scheme seems to have changed from X - X to X.X
+#   - [x] sourcetype no longer holds detection name but instead is stash_common_detection_model
+#   - [x] action.escu.full_search_name no longer available
+#   - [x] check to see if we can get "name"
+#   - [ ] move strings to enums
+#   - [x] additionally, timeout for cms_parser seems to need more time
+#   - [ ] validate multi-line fields -> search, description, action.notable.param.rule_description,
+#         action.notable.param.drilldown_searches
+
+# TODO (cmcginley): suppress logging
+# Suppress logging by default; enable for local testing
+ENABLE_LOGGING = True
+LOG_LEVEL = logging.DEBUG
+LOG_PATH = "content_versioning_service.log"
+
+# TODO (cmcginley): would it be better for this to only run on one instance? Or to consolidate
```

Comment:
Currently, validation of `cms_main` occurs as part of the instance setup for every instance during integration testing. This is not strictly necessary, as, if versioning is enabled properly on one instances, it presumably is setup appropriately on all other instances. We don't lose anything by running this validation in parallel, as we want to wait to begin testing until this step is complete whether it's being run on one instance or 40.

Any thoughts @pyth0n1c ?

Reply:
I am fine with doing this on every instance.  Right now we do make the very intentional assumption that all instances have the same versions of everything (apps. splunk core, etc). However, I could see in the future setting up multiple different versions configurations with test_servers and wanting to run validation on all of them.